crates/store/re_chunk_store/src/query.rs

```diff
@@ -75,6 +118,110 @@ impl ChunkStore {
+            .static_chunk_ids_per_entity
+            .get(entity_path)
+            .and_then(|static_chunks_per_component| static_chunks_per_component.get(component_name))
+            .and_then(|id| self.chunks_per_chunk_id.get(id))
+            .is_some()
+        {
+            return true;
+        }
+
+        for temporal_chunk_ids_per_component in self
+            .temporal_chunk_ids_per_entity_per_component
+            .get(entity_path)
+            .iter()
+            .flat_map(|temporal_chunk_ids_per_timeline| temporal_chunk_ids_per_timeline.values())
+        {
+            if temporal_chunk_ids_per_component
+                .get(component_name)
+                .is_some()
+            {
+                return true;
+            }
+        }
+
+        false
+    }
+
+    /// Check whether an entity has a specific component in its static data.
+    #[inline]
+    pub fn entity_has_static_component(
+        &self,
+        entity_path: &EntityPath,
+        component_name: &ComponentName,
+    ) -> bool {
+        re_tracing::profile_function!();
+
+        let Some(static_components) = self.static_chunk_ids_per_entity.get(entity_path) else {
+            return false;
+        };
+
+        static_components.contains_key(component_name)
+    }
+
+    /// Check whether an entity has any data on a specific timeline, or any static data.
+    pub fn entity_has_data_on_timeline(
+        &self,
+        timeline: &Timeline,
+        entity_path: &EntityPath,
+    ) -> bool {
+        re_tracing::profile_function!();
+
+        self.query_id.fetch_add(1, Ordering::Relaxed);
+
+        if self.static_chunk_ids_per_entity.get(entity_path).is_some() {
+            // Static data exists on all timelines
+            return true;
+        }
+
+        if let Some(temporal_chunk_ids_per_timeline) =
+            self.temporal_chunk_ids_per_entity.get(entity_path)
+        {
+            if temporal_chunk_ids_per_timeline.contains_key(timeline) {
+                return true;
+            }
+        }
+
+        false
+    }
+
+    /// Check whether an entity has any data on any timeline, or any static data.
+    pub fn entity_has_any_component_on_any_timeline(&self, entity_path: &EntityPath) -> bool {
+        re_tracing::profile_function!();
+
+        self.query_id.fetch_add(1, Ordering::Relaxed);
+
+        if let Some(static_chunks_per_component) = self.static_chunk_ids_per_entity.get(entity_path)
+        {
+            if !static_chunks_per_component.is_empty() {
+                return true;
+            }
+        }
+
+        if let Some(temporal_chunks_per_timeline) = self
+            .temporal_chunk_ids_per_entity_per_component
+            .get(entity_path)
+        {
+            for temporal_chunks_per_component in temporal_chunks_per_timeline.values() {
+                if !temporal_chunks_per_component.is_empty() {
+                    return true;
+                }
+            }
+        }
+
+        false
```

Comment:
I answered the 2nd question in a different thread, I do believe it is important to differentiate between having a component and actually having data.

As for style, I'll change this to use combinators instead. I agree that they're easier to read, and it's what the APIs here all use already.
I just default to writing imperative code, because it gives the compiler way less work to do. Those nested types are nasty for compile times and often mysteriously confuse LLVM.